### PR TITLE
chore: add GitHub action for stale issues and PRs

### PR DESCRIPTION
**Pull request by yizha1**
This PR enables the `specifications` repo to run stale action at 1:30 every day to label or close stale PRs and issues. See guideline https://github.com/marketplace/actions/close-stale-issues

This is the definition for stale PRs or issues that we discussed during community call and to be updated in [contributing PR](https://github.com/notaryproject/.github/pull/25).

"A stale issue is one that remains inactive or without updates for a period of 60 days. A stale pull request (PR) is one that remains inactive or without updates for a period of 45 days. When an issue or PR becomes stale, it is labelled as `stale`. Normally maintainers will comment on stale issues or PRs to prompt participants to take action. If there is no activity for additional 30 days, this issue or PR will be closed. If an update/comment occur on stale issues or pull requests, the stale label will be removed, and the timer will restart"

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

**Review by priteshbandi — APPROVED** [review]
LGTM

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - uses: actions/stale@v8
+        with:
+          stale-issue-message: "This issue is stale because it has been open 60 days with no activity. Remove stale label or comment or this will be closed in 30 days."
+          stale-pr-message: "This PR is stale because it has been open 45 days with no activity. Remove stale label or comment or this will be closed in 30 days."
+          close-issue-message: "This issue was closed because it has been stalled for 30 days with no activity."
+          close-pr-message: "This PR was closed because it has been stalled for 30 days with no activity."
+          days-before-issue-stale: 60
+          days-before-pr-stale: 45
+          days-before-issue-close: 30
+          days-before-pr-close: 30
```
Comment (whalelines) [inline .github/workflows/stale.yml:32]: Since these values are the same, you could just use `days-before-close: 30`

**Review by whalelines — COMMENTED** [review]
One minor suggestion

**Review by vaninrao10 — APPROVED** [review]
LGTM

File: .github/workflows/stale.yml (hunk 1 above)
Comment (yizha1) [reply .github/workflows/stale.yml:32]: Thank you @whalelines. My preference is to use distinct parameters for issues and PRs, allowing for independent updates in the future.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**